### PR TITLE
Sprint 51.1

### PR DESCRIPTION
**Pull request by ericsorenson**

### CODE REVIEW

File: get-es-instance-type.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ elif [[ $BRANCH == 'test' ]] ; then
 elif [[ $BRANCH == 'migration' ]] ; then
   echo "t2.small.elasticsearch"
 elif [[ $BRANCH == 'master' ]] ; then
-  echo "t2.small.elasticsearch"
+  echo "m5.large.elasticsearch"
```
Comment (mmarcotte) [inline get-es-instance-type.sh:32]: noted; FYI, we will be amending this for the following branches:

* test
* migration
* prod

File: shared/src/business/useCases/getDownloadPolicyUrlInteractor.js [hunk 2]
```diff
@@ -45,102 +45,119 @@ exports.getDownloadPolicyUrlInteractor = async ({
...
         throw new UnauthorizedError(
-          'Unauthorized to view document at this time',
+          'Unauthorized to view document at this time.',
         );
       } else {
         if (!userAssociatedWithCase) {
           throw new UnauthorizedError('Unauthorized');
         }
       }
     }
```
Comment (mmarcotte) [inline shared/src/business/useCases/getDownloadPolicyUrlInteractor.js:160]: This was some thoughtful work for access control ☝️ 👏

File: shared/src/business/useCases/public/getPublicDownloadPolicyUrlInteractor.js [hunk 3]
```diff
@@ -24,12 +24,25 @@ exports.getPublicDownloadPolicyUrlInteractor = async ({
...
   });
 
+  if (!docketEntryEntity) {
+    throw new NotFoundError(`Docket entry ${key} was not found.`);
+  }
+  if (!docketEntryEntity.isFileAttached) {
+    throw new NotFoundError(
+      `Docket entry ${key} does not have an attached file.`,
+    );
+  }
```
Comment (mmarcotte) [inline shared/src/business/useCases/public/getPublicDownloadPolicyUrlInteractor.js:44]: Love seeing all of these intentional errors being thrown. 💪

File: shared/src/persistence/dynamo/cases/docketNumberGenerator.js [hunk 4]
```diff
@@ -24,5 +23,23 @@ exports.createDocketNumber = async ({ applicationContext, receivedAt }) => {
...
+      pk: `case|${docketNumber}`,
+      sk: `case|${docketNumber}`,
+    },
+    applicationContext,
+  });
+
+  if (caseMetadata) {
+    // be sure case with this docket number doesn't already exist -- if it does, stop!
+    const message = 'docket number already exists!';
+    applicationContext.logger.info(message, docketNumber);
```
Comment (mmarcotte) [inline shared/src/persistence/dynamo/cases/docketNumberGenerator.js]: can this be logger.error( ... )

**Comment by julialeague** [comment]
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] If changes were made in the UI (if not, skip):
  - [x] If desired, run `docker system prune` to remove any unused docker images from previous code review.
  - [x] Fetch the pull request for the sprint (e.g., `git fetch origin pull/{PR #}/head:sprint-{Sprint #}`), and then switch to that branch (e.g. `git checkout sprint-{Sprint #}`)
  - [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
	- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
	- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
	- [x] test in Mobile Safari and Mobile Chrome (or an emulator like Chrome DevTools), with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
	- [x] user-visible changes (including API users like the IRS) are documented in CHANGELOG.md (which is linked from US Tax Court website).
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines

- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: web-api/src/lambdaWrapper.js [hunk 5]
```diff
@@ -16,14 +16,26 @@ export const lambdaWrapper = lambda => {
       body: JSON.stringify(req.body),
     });
     res.status(response.statusCode);
+
+    res.set({
+      'Access-Control-Allow-Origin': '*',
```
Comment (mmarcotte) [inline web-api/src/lambdaWrapper.js:21]: This doesn't look like a change in behavior, but need to be sure we are using this properly. 🤔

File: web-client/src/views/DocketRecord/DocumentViewer.jsx [hunk 6]
```diff
@@ -47,7 +47,7 @@ export const DocumentViewer = connect(
                           viewDocumentId === entry.docketEntryId && 'active',
                         )}
                         isActive={viewDocumentId === entry.docketEntryId}
-                        key={idx}
+                        key={entry.docketEntryId}
```
Comment (mmarcotte) [inline web-client/src/views/DocketRecord/DocumentViewer.jsx:50]: nice; @ericsorenson once told me uuids were fairly unique 😉

**Review by mmarcotte — COMMENTED** [review]
This is looking really clean! Great work out there. 💪 💯 My only real worry is the CORS header declaration. Might need to live as a Backlog consideration for now.

File: shared/src/persistence/dynamo/cases/docketNumberGenerator.js (hunk 4 above)
Comment (ericsorenson) [reply shared/src/persistence/dynamo/cases/docketNumberGenerator.js]: Updated!

File: web-api/src/lambdaWrapper.js (hunk 5 above)
Comment (ericsorenson) [reply web-api/src/lambdaWrapper.js:21]: We can certainly lock this down to the web client's URLs.

Comment (mmarcotte) [reply web-api/src/lambdaWrapper.js:21]: @ericsorenson -- can you log an item to revisit this?

Comment (ericsorenson) [reply web-api/src/lambdaWrapper.js:21]: You bet.

**Review by mmarcotte — APPROVED** [review]

File: web-api/src/lambdaWrapper.js (hunk 5 above)
Comment (ericsorenson) [reply web-api/src/lambdaWrapper.js:21]: https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/6918

**Pull request merged**